### PR TITLE
Register gateway: Provide macros for the shared bit

### PR DESCRIPTION
**Pull request by AlessandroA**
@meriac @patater @niklas-arm 

The API would now look like this:

```C
uint16_t val = uvisor_read(my_box, UVISOR_RGW_EXCLUSIVE, address, UVISOR_RGW_OP_READ, 0xFFFFFFFFUL);
```

### CODE REVIEW

**Comment by Patater** [comment]
LGTM

File: api/inc/register_gateway.h [hunk 1]
```diff
@@ -143,77 +145,95 @@
...
+ *                     not. Two values are available: UVISOR_RGW_SHARED,
+ *                     UVISOR_RGW_EXCLUSIVE.
  * @param address[in]  Address at which to check the bits
  * @param mask[in]     Bits to select out of the target address
  * @returns The value `(bool) (*address & mask) == mask)`.
  */
-#define UVISOR_BITS_CHECK(box_name, address, mask) \
-    ((bool) (UVISOR_BITS_GET(box_name, address, mask) == mask))
+#define UVISOR_BITS_CHECK(box_name, shared, address, mask) \
+    ((bool) (UVISOR_BITS_GET(box_name, shared, address, mask) == mask))
```
Comment (niklas-arm) [inline api/inc/register_gateway.h]: Is `bool` acceptable here? The user would have to have `#include <stdbool.h>` somewhere, but we cannot guarantee this, unless we include it here as well.

**Comment by niklas-arm** [comment]
👍  for moving the shared bit forward.

**Comment by niklas-arm** [comment]
Apart from `bool` usage, LTGM.

File: api/inc/register_gateway.h (hunk 1 above)
Comment (AlessandroA) [inline api/inc/register_gateway.h]: Good point! I removed the `bool`. The macro can still effectively be used in conditional statements.

**Comment by AlessandroA** [comment]
* Addressed @niklas-arm's comments.

**Pull request merged**